src/main/java/duke/handlers/InputParser.java

```diff
@@ -0,0 +1,119 @@
+package duke.handlers;
+
+import duke.definitions.*;
+import duke.exceptions.EmptyContentException;
+import duke.exceptions.InvalidParamArgument;
+
+import java.util.HashMap;
+
+public final class InputParser {
+    private String input;
+
+    public InputParser(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Example input: deadline do homework /by tomorrow /note skip page 70
+     * commandString: "deadline"
+     * CommandPacket created:
+     * {
+     *  commandType: ADD_DEADLINE
+     *  commandContent: "do homework"
+     *  params: HashMap<String, String>
+     *  {
+     *   "by": "tomorrow"
+     *   "note": "skip page 70"
+     *  }
+     * }
+     */
```

Comment:
Comprehensive comment that illustrates why the method is defined as such. However, may consider adapting them to follow Javadoc conventions (which will be taught in week 7).